lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -1009,8 +954,7 @@ void ReabstractionInfo::specializeConcreteAndGenericSubstitutions(
     auto CanTy = GP->getCanonicalType();
     Type Replacement = CalleeGenericSig->getConcreteType(CanTy, *SM);
     if (!Replacement)
-      Replacement = CalleeInterfaceToCallerArchetypeMap.lookupSubstitution(
-          cast<SubstitutableType>(CanTy));
+      Replacement = CanTy.subst(CalleeInterfaceToCallerArchetypeMap);
```

Comment:
What do you mean here? `CanTy`? It is done because later it is used for `getConformances` which require CanType.